docs-v2/understand/architecture.mdx

```diff
@@ -4,6 +4,77 @@ sidebarTitle: 'Architecture'
 description: 'Learn more about the architecture of Nango.'
 ---
 
-<Info>
-The docs recently underwent a makeover, leaving this page empty. We are working hard to fill in all the missing pages within the next two weeks. Meanwhile, ask any questions on the [community](https://nango.dev/slack).
-</Info>
\ No newline at end of file
+Nango improves the way applications integrate with external APIs by providing a flexible, customizable unified API. Unlike traditional unified APIs that offer a one-size-fits-all solution, Nango empowers developers to tailor the interaction between their app and external APIs, ensuring a perfect fit for their specific needs.
+
+# Core components
+
+- **[Integrations](/understand/concepts/integrations):** The configurations that define how to connect to an external API. Each integration represents a separate external service (like Hubspot or Slack) and includes details such as authentication methods and consumable API endpoints.
+
+- **[Connections](/understand/concepts/connections):** Connections are instances of integrations for individual users or entities. They store the authentication credentials (like OAuth tokens or API keys) that allow Nango to access the external API on behalf of a specific user or entity.
+
+- **[Scripts](/understand/concepts/scripts):** Scripts are the heart of Nango's customization capabilities. They define the logic for how Nango interacts with external APIs, and run on Nango's infrastructure. Scripts are either provided by Nango as [templates](/understand/concepts/templates) or implemented by you. There are different types of scripts:
+
+    - **[Action](/understand/concepts/actions) scripts:** Enable specific tasks (e.g., creating a contact) to be executed with an external API.
+
+    - **[Sync](/understand/concepts/syncs) scripts:** Automate the synchronization of data from external APIs into your app.
+
+    - **[Webhook](/understand/concepts/webhooks) scripts:** Handle incoming webhooks from external APIs for real-time updates.
```

Comment:
Action scripts, Sync scripts, and Webhook scripts seems a bit repetitive

Reply:
Agreed, will change!